iamctl/pkg/utils/setup.go

```diff
@@ -0,0 +1,188 @@
+	TenantDomain string `json:"TENANT_DOMAIN"`
+	Username     string `json:"USERNAME"`
+	Password     string `json:"PASSWORD"`
+	Token        string `json:"TOKEN"`
+}
+
+type ToolConfigs struct {
+	KeywordMappings    map[string]interface{} `json:"KEYWORD_MAPPINGS"`
+	ApplicationConfigs map[string]interface{} `json:"APPLICATIONS"`
+}
+
+type Application struct {
+	Id          string `json:"id"`
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	Self        string `json:"self"`
+}
+
+type AppList struct {
+	TotalResults int           `json:"totalResults"`
+	StartIndex   int           `json:"startIndex"`
+	Count        int           `json:"count"`
+	Applications []Application `json:"applications"`
+	Links        []string      `json:"links"`
+}
+
+var SERVER_CONFIGS ServerConfigs
+var TOOL_CONFIGS ToolConfigs
+
+func LoadConfigs(envConfigPath string) (baseDir string) {
+
+	baseDir = filepath.Dir(filepath.Dir(envConfigPath))
+	serverConfigFile := filepath.Join(envConfigPath, "serverConfig.json")
+	toolConfigFile := filepath.Join(envConfigPath, "toolConfig.json")
+
+	// Load configs from files.
+	SERVER_CONFIGS = loadServerConfigsFromFile(serverConfigFile)
+	TOOL_CONFIGS = loadToolConfigsFromFile(toolConfigFile)
+
+	return baseDir
+}
+
+func loadServerConfigsFromFile(configFilePath string) (serverConfigs ServerConfigs) {
+
+	configFile, err := os.Open(configFilePath)
+	if err != nil {
+		log.Fatalln(err.Error())
+	}
+	defer configFile.Close()
+
+	jsonParser := json.NewDecoder(configFile)
+	err = jsonParser.Decode(&serverConfigs)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	fmt.Println("Server configs loaded succesfully from the config file.")
+
+	// Set tenant domain if not defined in the config file.
+	if serverConfigs.TenantDomain == "" {
+		log.Println("Tenant domain is not defined in the config file. Using the default tenant domain: carbon.super")
+		serverConfigs.TenantDomain = "carbon.super"
+	}
+
+	// Get access token.
+	serverConfigs.Token = getAccessToken(serverConfigs)
+	fmt.Println("Access Token recieved succesfully.")
+
+	return serverConfigs
+}
+
+func loadToolConfigsFromFile(configFilePath string) (toolConfigs ToolConfigs) {
+
+	configFile, err := ioutil.ReadFile(configFilePath)
+	if err != nil {
+		log.Fatalln("Error when reading the tool config file.", err.Error())
+	}
+
+	if len(configFile) == 0 {
+		return toolConfigs
+	}
+
+	err = json.Unmarshal(configFile, &toolConfigs)
+	if err != nil {
+		log.Fatalln("Tool configs are not in the correct format. Please check the config file.", err)
+	}
+
+	log.Println("Tool configs loaded successfully from the config file.")
+	return toolConfigs
+}
+
+func getAccessToken(config ServerConfigs) string {
+
+	var err error
+	var response oAuthResponse
+
+	if config.ServerUrl == "" {
+		log.Fatalln("Server URL is not defined in the config file.")
+	}
+	authUrl := config.ServerUrl + "/oauth2/token"
+
+	body := url.Values{}
+	body.Set("grant_type", "password")
```

Comment:
Any specific reason for using `password` grant here? Usually password grant is not recommended as username, password needs to be provided.